docs/en/integrations/build-integration.asciidoc

```diff
@@ -208,20 +208,88 @@ apache
+│   │          base-fields.yml
+│   │          ecs.yml
+│   │          fields.yml
 │   └───error
 │   └───status
 ----
-<1> `agent.yml` fields the Elastic agent uses
-<2> `base-fields.yml` never changes and is required for all integrations
-<3> Defines the relevant ECS fields
-<4> Custom Apache access log fields
 
+=== agent.yml
+The `agent.yml` file defines fields used by default processors, this file usually remains unchanged between integrations.
+
+=== base-fields.yml
+The fields defined in this file are placeholders for the datastream subfields which, in turn, receive their value from the Elastic agent.
+In addition, the fields `event.module` and `event.dataset` are also present; the values in these fields depend on the integration.
+For example, in the Apache integration (more specifically, for the access dataset) that would be:
+* event.module: apache
+* event.dataset: apache.access
+
+=== ecs.yml
+This file specifies every Elastic Common Schema (ECS) field used by the integration that is not defined in the files `agent.yml` or `base-fields.yml` files.
+To avoid redefining every ECS field in every integration, we can specify an existing field as external. For example:
+
+[source,yaml]
+----
+- external: ecs
+  name: client.ip
+- external: ecs
+  name: destination.domain
+----
+
+The `external: ecs` definition instructs the `elastic-package` command line tool to refer to an external ECS reference to resolve specific fields. By default it looks at the https://raw.githubusercontent.com/elastic/ecs/v8.6.0/generated/ecs/ecs_nested.yml[ECS reference] file hosted on Github. This external reference file is determined by a Git reference found in the `_dev/build/build.yml` file, located starting from the root of the package directory.
+The `build.yml` file should look like the following:
++
+[source,yaml]
+----
+dependencies:
+  ecs:
+    reference: git@v8.6.0
+----
+
+In addition to this approach, we support three more methods to import the ECS, described below.
+
+. Dynamic mapping import
+Adding `import_mappings: true` to the ECS section in `_dev/build/build.yml` introduces a https://github.com/elastic/elastic-package/blob/f439b96a74c27c5adfc3e7810ad584204bfaf85d/internal/builder/_static/ecs_mappings.yaml[dynamic mapping] with most of the ECS definitions. Using this method means ECS fields don't need to be defined in your integration, it is dynamically integrated into the package at build time.
+Adjusting the aformentioned `build.yml` file for this method:
++
+[source,yaml]
+----
+dependencies:
+  ecs:
+    reference: git@v8.6.0
+    import_mappings: true
+----
+
+. Fleet pre-installed template (from version 8.13.0)
+Starting with version 8.13.0, Fleet automatically pre-installs the `ecs@mappings` component template containing the ECS definitions into all integrations. This template is similar to the dynamic mapping method, apart from the fact that it does so on ingestion instead of including the mappings into the package.
```

Comment:
Maybe we should put this point first, mentioning that if the package supports only versions starting on 8.13.0, it doesn't require to use any other method, except for corner cases. More details in https://github.com/elastic/observability-docs/pull/3792/files#r1588831495